src/main/java/uk/gov/hmcts/divorce/document/DocumentGenerationUtil.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.hmcts.divorce.document;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.document.model.DocumentType;
+
+import java.util.List;
+
+import static org.apache.commons.collections4.CollectionUtils.isEmpty;
+import static uk.gov.hmcts.divorce.document.DocumentConstants.CERTIFICATE_OF_ENTITLEMENT_JUDICIAL_SEPARATION_TEMPLATE_ID;
+import static uk.gov.hmcts.divorce.document.DocumentConstants.CERTIFICATE_OF_ENTITLEMENT_NAME;
+import static uk.gov.hmcts.divorce.document.DocumentConstants.CERTIFICATE_OF_ENTITLEMENT_TEMPLATE_ID;
+import static uk.gov.hmcts.divorce.document.model.DocumentType.CERTIFICATE_OF_ENTITLEMENT;
+
+@RequiredArgsConstructor
+@Component
+public class DocumentGenerationUtil {
```

Comment:
Do we need an extra util class? Feels like `removeExistingDocuments` could be in the existing DocumentUtils class

Reply:
Removed extra class and moved methods into relevant classes